### PR TITLE
prevent sending funds to spent address +ui

### PR DESCRIPTION
**Pull request by alon-e**
show warning for sending to spent addresses
rename filterSpentInputs() -> filterSpentAddresses()
added UI elements.
(includes #270 #271 changes)

### CODE REVIEW

File: locales/en/translation.json [hunk 1]
```diff
@@ -311,8 +311,10 @@
   "request_error":"Request Error",
   "invalid_transaction_hash": "Invalid Transaction Hash",
   "seed_character_set": "Only uppercase latin letters and the number 9 are allowed.",
-  "double_spend_warning": "Double Spend Warning",
-  "double_spend_attempt": "You are attempting to spend an already used input. You should wait for your transaction to be confirmed first before attempting to make a conflicting transaction. Do you want to continue anyway?",
+  "key_reuse_error": "Error: Private Key Reuse Detected!",
+  "key_reuse_attempt": "You are attempting to sign a transaction with input that have an already been used. IOTA uses Winternitz one-time signature (W-OTS) scheme, due to it's `one-time` nature, the security of funds in an address decreases rapidly if you re-sign new transactions using the same key.Please wait for previous transaction to confirm, before sending another transaction.",
```
Comment (th0br0) [inline locales/en/translation.json]: missing whitespace after `key.`
also, we should request them to reattach `... to confirm, e.g. by reattaching, before sending ...`

File: locales/en/translation.json [hunk 2]
```diff
@@ -311,8 +311,10 @@
...
+  "sent_to_key_reuse_error": "Error: Sending to a used address!",
+  "sent_to_key_reuse_attempt": "You are attempting to send a transaction to an address that has already been used. Please ask recipient for a new address, that hasn't been spent from yet.",
```
Comment (th0br0) [inline locales/en/translation.json]: `that has already been used (i.e. tokens have already been spent from this address).`

File: ui/js/ui.transfers.js [hunk 3]
```diff
@@ -81,7 +81,18 @@ var UI = (function(UI, $, undefined) {
...
+        var exptectedOutputsLength = outputsToCheck.length;
+        filterSpentAddresses(outputsToCheck).then(filtered => {
+          if (filtered.length !== exptectedOutputsLength) {
+            UI.isDoingPOW = false;
+            UI.formError("transfer", "sent_to_key_reuse_error", {"initial": "send_it_now"});
+            modal = $("#sent-to-key-reuse-modal").remodal({hashTracking: false, closeOnOutsideClick: false, closeOnEscape: false});
+            modal.open();
+            return;
+          }
+          iota.api.sendTransfer(connection.seed, connection.depth, connection.minWeightMagnitude, transfers, {"inputs": inputs.inputs}, function(error, transfers) {
```
Comment (th0br0) [inline ui/js/ui.transfers.js]: inputs is not defined? or is that a global?

File: locales/en/translation.json (hunk 2 above)
Comment (th0br0) [reply locales/en/translation.json]: similar change should probably happen to `sent_to_key_reuse_error` as well.

File: ui/js/ui.transfers.js (hunk 3 above)
Comment (chrisdukakis) [reply ui/js/ui.transfers.js]: https://github.com/alon-e/wallet/blob/5c8c6e336336abbe959f521caa346ce6f258891b/ui/js/ui.transfers.js#L65
It's private in callback

Comment (alon-e) [reply ui/js/ui.transfers.js]: it's the callback param from `getUnspentInputs`:L65

File: locales/en/translation.json (hunk 1 above)
Comment (alon-e) [reply locales/en/translation.json]: done.

File: locales/en/translation.json (hunk 2 above)
Comment (alon-e) [reply locales/en/translation.json]: done.

**Pull request merged**